Cpp/odin-views/permission.cpp

```diff
@@ -27,48 +101,48 @@ namespace {
                 const fostlib::string &path,
                 fostlib::http::server::request &req,
                 const fostlib::host &host) const {
-            const auto &permission = config["permission"];
-            bool granted = false;
-            if (odin::c_jwt_trust.value() && req.headers().exists("__jwt")) {
-                const auto &perm_header = req.headers()["__jwt"].subvalue(
-                        odin::c_jwt_permissions_claim.value());
-                if (perm_header) {
-                    auto perms = fostlib::json::parse(perm_header.value());
-                    for (const auto &p : perms) {
-                        if (p == permission) {
-                            granted = true;
-                            break;
+            return check_permission(config, config, path, req, host);
+        }
+    } c_permission;
+
+
+    const class permission_method : public fostlib::urlhandler::view {
+      public:
+        static const fostlib::jcursor userloc;
+
+        permission_method() : view("odin.permission.method") {}
+
+        std::pair<boost::shared_ptr<fostlib::mime>, int> operator()(
+                const fostlib::json &config,
+                const fostlib::string &path,
+                fostlib::http::server::request &req,
+                const fostlib::host &host) const {
+
+            if (req.method() == "HEAD" and config.has_key("GET")) {
+                // HEAD method always use the same permission as GET
+                return check_permission(config["GET"], config, path, req, host);
+            } else if (config.has_key(req.method())) {
+                return check_permission(config[req.method()], config, path, req, host);
+            } else {
+                if (config.has_key("otherwise")) {
+                    return fostlib::urlhandler::view::execute(
+                        config["otherwise"], path, req, host);
+                } else {
+                    auto otherwise_config = fostlib::json();
+                    // Default view for otherwise is method not allowed
+                    fostlib::insert(otherwise_config, "view", "fost.response.405");
+                    // Take all allowed methods from configured verbs
+                    for (auto c = config.begin(); c != config.end(); ++c) {
+                        if (c.key() != "otherwise") {
+                            fostlib::push_back(otherwise_config, "configuration", "allow", c.key());
                         }
                     }
+                    return fostlib::urlhandler::view::execute(
```

Comment:
Probably before this we should add `HEAD` if `GET` is there. Hmmm, probably that should be done by the 405 implementation instead of everywhere that writes the config for it... So this is fine as it is.